src/main/java/org/jboss/modules/ResourceLoaders.java

```diff
@@ -75,10 +77,21 @@ public static IterableResourceLoader createIterableFileResourceLoader(final Stri
      * @param jarFile the backing JAR file
      * @return the resource loader
      */
+    public static ResourceLoader createJarResourceLoader(final String name, final Path jarFile) throws IOException {
+        return new JarResourceLoader(name, jarFile, AccessController.getContext());
+    }
+
+    @Deprecated
     public static ResourceLoader createJarResourceLoader(final String name, final JarFile jarFile) {
-        return new JarFileResourceLoader(name, jarFile);
```

Comment:
@ctomc - is possible to do some reasonable performance comparison of  old JarFileResourceLoader vs. new JarResourceLoader? Thinking what would do the difference I noticed and the loader class is my tip...